backend/app/repositories/firestore.py

```
@@ -448,6 +508,81 @@ def do_cancel(transaction) -> Optional[Dict[str, Any]]:
+            Atomically create a batch of slot bookings when capacity and uniqueness constraints allow it.
+            
+            Returns:
+            	List[Dict[str, Any]]: The created booking records.
+            	None: If the batch exceeds capacity or any booking or active-slot record already exists.
+            """
+            counter_doc = counter_ref.get(transaction=transaction)
+            current = counter_doc.to_dict().get("count", 0) if counter_doc.exists else 0
+            if current + len(booking_ids) > capacity:
+                return None
+
+            for bref, aref in zip(booking_refs, active_refs):
```

Ruff (0.16.3)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)